Zend/tests/gh13142.phpt

```diff
@@ -0,0 +1,15 @@
+--TEST--
+GH-13142 (Undefined variable name is shortened when contains \0)
+--FILE--
+<?php
+
+$a = "test\0test";
```

Comment:
It might be better if there were also tests in the case of `\0test` and `test\0`